src/py21cmfast/src/SpinTemperatureBox.c

```diff
@@ -101,9 +123,27 @@ if (LOG_LEVEL >= DEBUG_LEVEL){
 
     int NO_LIGHT = 0;
 
+    if(flag_options->USE_MASS_DEPENDENT_ZETA) {
+        ION_EFF_FACTOR = global_params.Pop2_ion * astro_params->F_STAR10 * astro_params->F_ESC10;
+        ION_EFF_FACTOR_MINI = global_params.Pop3_ion * astro_params->F_STAR7_MINI * astro_params->F_ESC7_MINI;
+    }
+    else {
+        ION_EFF_FACTOR = astro_params->HII_EFF_FACTOR;
+		ION_EFF_FACTOR_MINI = 0.;
+    }
+
     // Initialise arrays to be used for the Ts.c computation //
     fftwf_complex *box = (fftwf_complex *) fftwf_malloc(sizeof(fftwf_complex)*HII_KSPACE_NUM_PIXELS);
     fftwf_complex *unfiltered_box = (fftwf_complex *)fftwf_malloc(sizeof(fftwf_complex)*HII_KSPACE_NUM_PIXELS);
+    fftwf_complex *log10_Mcrit_LW_unfiltered, *log10_Mcrit_LW_filtered;
+    if (flag_options->USE_MINI_HALOS){
+        log10_Mcrit_LW_unfiltered = (fftwf_complex *) fftwf_malloc(sizeof(fftwf_complex)*HII_KSPACE_NUM_PIXELS);
+        log10_Mcrit_LW_filtered = (fftwf_complex *) fftwf_malloc(sizeof(fftwf_complex)*HII_KSPACE_NUM_PIXELS);
+        log10_Mcrit_LW = (float **) calloc(global_params.NUM_FILTER_STEPS_FOR_Ts,sizeof(float *));
```

Comment:
These do not appear to be free'd at the end of the file. Please make sure all declared memory is free'd.

Reply:
It's also worth running a test script through once with `valgrind` to check for these kinds of errors.